nearcore/src/cold_storage.rs

```diff
@@ -140,30 +172,32 @@ fn sanity_check_impl(
     cold_head_height: u64,
     hot_final_head_height: u64,
     hot_tail_height: u64,
-) -> anyhow::Result<()> {
+) -> anyhow::Result<(), ColdStoreError> {
     // We should only copy final blocks to cold storage.
     if cold_head_height > hot_final_head_height {
-        return Err(anyhow::anyhow!(
-            "Cold head is ahead of final head. cold head height: {} final head height {}",
+        return Err(ColdStoreError::ColdHeadAheadOfFinalHeadError {
             cold_head_height,
-            hot_final_head_height
-        ));
+            hot_final_head_height,
+        });
     }
+
     // Cold and Hot storages need to overlap. Without this check we would skip
     // blocks from cold_head_height to hot_tail_height. This will result in
     // corrupted cold storage.
     if cold_head_height < hot_tail_height {
-        return Err(anyhow::anyhow!(
-            "Cold head is behind hot tail. cold head height: {} hot tail height {}",
+        return Err(ColdStoreError::ColdHeadBehindHotTailError {
             cold_head_height,
-            hot_tail_height
-        ));
+            hot_tail_height,
+        });
     }
     Ok(())
 }
 
-fn cold_store_copy_result_to_string(result: &anyhow::Result<ColdStoreCopyResult>) -> &str {
+fn cold_store_copy_result_to_string(
+    result: &anyhow::Result<ColdStoreCopyResult, ColdStoreError>,
+) -> &str {
     match result {
+        Err(ColdStoreError::ColdHeadBehindHotTailError { .. }) => "cold_head_behind_hot_tail_error",
```

Comment:
What about other ColdStoreErrors?

Reply:
They can be easily added here. I only had an interest in exposing this particular error and the rest will go under the generic `error`

Reply:
I will add the other errors here as well